patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+
+Build up agreed security requirement of InnerSource like:
+
+- No hard-coded usernames and/or passwords.
+- Repo does not contain client information.
+- Business critical algorithms may not be placed in SHARED repositories.
+- Code from SHARED repositories will not be distributed directly to Production.
+- Leveraging secrets scanning tools to scan the target repositories (including code, test cases, and helper scripts) for confidential data such as accounts, passwords, access tokens, keys, and other sensitive data.
+- Keep repositories CLOSED, but expose metadata about the projects (i.e. through an [InnerSource Portal][innersource-portal]), and create some kind of access request workflow. This way you could still give people access to the code, but not open it to everyone by default. *(Note: The full model is described under **Extension: An Additional Sharing Level**)*
+
+### Extension: An Additional Sharing Level
+
+In some cases, introducing additional sharing levels might be appropriate. Use cases include semi-sensitive code that can be shared with larger groups of people, but for example not with all interns and contractors. Or not with people with certain nationalities or working in a specific country, in order to comply with [export control][export-control] regulations. Or legacy repositories that are not (yet) ready to be made available as INTERNAL. Without the additional level, (too) many repositories will end up as CLOSED in an existing organization, as all businesses will considering their code 'special', requiring extra protection and/or 'not ready for opening up company wide yet (e.g. hard-coded secrets, but also technical debt, lacking documentation).
+
+And given 'findability' is also important, adding the additional level that facilitates findability, but still provides a light-weight gate seems to be a good way out to steer projects out of closed source.
+
+As concrete example, which allows repositories to be discovered, even though there is no company-wide access, consider the following sharing levels:
+
+- **PUBLIC** - open source: findable and accessible for all SW developers in the world
+- **INTERNAL** - inner source: findable and accessible for all SW developers in the company
+- **RESTRICTED** - restricted source: findable for all SW developers in the company, low-friction process to gain full access.
+- **CLOSED** - closed source: only accessible to named individuals and not findable via the search engine for other SW developers in the company.
+
+Repositories with RESTRICTED sharing level are included in the [central catalog][innersource-portal] and/or fed into the central search mechanism, together with their `README.md` and some meta information. Their content, however, is not accessible without following a light-weight access request process. This can for example be contacting one of the code owners or some other request mechanism described in the `README.md` file.
```

Comment:
```suggestion
Repositories with RESTRICTED sharing level are included in the [central catalog][innersource-portal] and/or fed into the central search mechanism, together with their `README.md` and some meta information. Their full content however, is not accessible without following a light-weight access request process. This can for example be contacting one of the code owners or some other request mechanism described in the `README.md` file.
```